### PR TITLE
RSDK-3055 - add viam-cpp-sdk to brew

### PR DESCRIPTION
**Pull request by stuqdog**

### CODE REVIEW

**Review by benjirewis — APPROVED** [review]
LGTM; did you [test locally](https://github.com/viamrobotics/homebrew-brews/tree/main?tab=readme-ov-file#development)?

**Comment by stuqdog** [comment]
> LGTM; did you [test locally](https://github.com/viamrobotics/homebrew-brews/tree/main?tab=readme-ov-file#development)?

I did! It successfully built and put the SDK in `/opt/homebrew/...`

File: Formula/viam-cpp-sdk.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  depends_on "protobuf"
+  depends_on "xtensor"
+  depends_on "pkg-config"
+  depends_on "ninja"
+  depends_on "buf"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
```
[the remark this answers is not in the document]
Comment (stuqdog) [reply Formula/viam-cpp-sdk.rb:23]: Yep! A slightly updated version of the `cmake` invocations, 
```cmake -S . -B build -G Ninja
cmake --build build --target all install```
is the new 
```cd build
cmake .. -G Ninja
ninja all && ninja install
``` 
:)

**Review by Otterverse — APPROVED** [review]
As long as it was tested on both mac architectures and on linux, then this looks good to go.

**Comment by stuqdog** [comment]
> As long as it was tested on both mac architectures and on linux, then this looks good to go.

Ah! I tested on my mac but that's it. Is there a preferred way for testing brew formulae on x86 mac/linux?

**Comment by Otterverse** [comment]
> > As long as it was tested on both mac architectures and on linux, then this looks good to go.
> 
> Ah! I tested on my mac but that's it. Is there a preferred way for testing brew formulae on x86 mac/linux?

For linux, you can fire up a docker image with like "debian:latest" as the image, install brew, and see if it works. (make sure `uname -a` reports x86_64, it might default to arm64 on mac's docker, not sure.)


You'd need a native mac for the other architecture there I think. That said, I'm less worried about the formula itself if it already works on your mac, as mac is very consistent. It'd only likely be code in the actual sdk code (or other code we produce that it uses) where I'd expect problems. So if you don't have an x86 mac handy, I'd still be okay with the merge as long as linux works.

**Comment by stuqdog** [comment]
Okay, this built in a brand new docker container. I had to do some preliminary installs (curl, git) to get brew installed; I assume this is fine to not account for. I also had to set `-j 3` in the docker container in the build job to avoid OOM sadness. Do we think this flag is worth putting in the formula in general? I haven't had the issue outside of docker. @Otterverse

**Comment by Otterverse** [comment]
> Okay, this built in a brand new docker container (edit: `ubuntu:latest`, to be explicit). I had to do some preliminary installs (curl, git) to get brew installed; I assume this is fine to not account for. I also had to set `-j 3` in the docker container in the build job to avoid OOM sadness. Do we think this flag is worth putting in the formula in general? I haven't had the issue outside of docker. @Otterverse

Sorry I didn't reply sooner. I didn't see this PR as "awaiting review" (since I'd technically approved it) so I lost track of it.

To answer, I think you're good to go. All that's expected. Someone has to have brew working. Docker resources aren't the same as the real world, and anyone installing on actual linux can just use their whole machine.

**Pull request merged**